Add support to rename clusters and cluster replicas. (#19642)
MaterializeInc/materialize: doc/user/content/sql/alter-rename.md

    ON v1.a = v2.a
    ```

> - Rename clusters and replicas if they are system clusters. For example, the `mz_introspection` cluster cannot be renamed.

Reviewer: Tweak to work with the list intro (`Your cannot rename items if:`):

```suggestion
- They are system clusters, such as `mz_system` and `mz_introspection`, or replicas in system clusters.
```
